diff --git a/target/linux/ipq40xx/base-files/etc/board.d/02_network b/target/linux/ipq40xx/base-files/etc/board.d/02_network
index 795f1d72c2..c8956ff68f 100644
--- a/target/linux/ipq40xx/base-files/etc/board.d/02_network
+++ b/target/linux/ipq40xx/base-files/etc/board.d/02_network
@@ -56,6 +56,7 @@ ipq40xx_setup_interfaces()
 		ucidef_set_interfaces_lan_wan "lan" "wan"
 		;;
 	aruba,ap-303|\
+	aruba,ap-305|\
 	aruba,ap-365|\
 	avm,fritzrepeater-1200|\
 	dlink,dap-2610|\
diff --git a/target/linux/ipq40xx/base-files/lib/upgrade/platform.sh b/target/linux/ipq40xx/base-files/lib/upgrade/platform.sh
index ed62eabe12..f0365e46d3 100644
--- a/target/linux/ipq40xx/base-files/lib/upgrade/platform.sh
+++ b/target/linux/ipq40xx/base-files/lib/upgrade/platform.sh
@@ -108,6 +108,7 @@ platform_do_upgrade() {
 	8dev,jalapeno|\
 	aruba,ap-303|\
 	aruba,ap-303h|\
+	aruba,ap-305|\
 	aruba,ap-365|\
 	avm,fritzbox-7530|\
 	avm,fritzrepeater-1200|\
diff --git a/target/linux/ipq40xx/image/generic.mk b/target/linux/ipq40xx/image/generic.mk
index 7738eca6e3..874b7d7dc3 100644
--- a/target/linux/ipq40xx/image/generic.mk
+++ b/target/linux/ipq40xx/image/generic.mk
@@ -189,6 +189,14 @@ define Device/aruba_ap-365
 endef
 TARGET_DEVICES += aruba_ap-365
 
+define Device/aruba_ap-305
+	$(call Device/aruba_glenmorangie)
+	DEVICE_MODEL := AP-305
+	DEVICE_PACKAGES := kmod-hwmon-ad7418 ath10k-firmware-qca99x0-ct
+	DEVICE_DTS_CONFIG := Glenmorangie@1
+endef
+TARGET_DEVICES += aruba_ap-305
+
 define Device/asus_map-ac1300
 	$(call Device/FitImageLzma)
 	DEVICE_VENDOR := ASUS
